### PR TITLE
Switch user roles to crm sdk

### PR DESCRIPTION
**Pull request by dawid-pawulski**

### CODE REVIEW

File: src/HE.InvestmentLoans.BusinessLogic/User/LoanUserContext.cs [hunk 1]
```diff
@@ -29,10 +27,9 @@ public class LoanUserContext : ILoanUserContext
 
     private UserAccount? _selectedAccount;
 
-    public LoanUserContext(IUserContext userContext, ILoanUserRepository loanUserRepository, ICacheService cacheService, IOrganizationServiceAsync2 service, IContactService contactService)
+    public LoanUserContext(IUserContext userContext, ICacheService cacheService, IOrganizationServiceAsync2 service, IContactService contactService)
```
Comment (lukasz-ciura-pwc) [inline src/HE.InvestmentLoans.BusinessLogic/User/LoanUserContext.cs:30]: LoanUserRepository should stay. IOrganizationServiceAsync2 should be used inside of LoanUserRepository.

The same thing with this https://github.com/homesengland/he-ssp/pull/221/files

**Review by lukasz-ciura-pwc — CHANGES_REQUESTED** [review]

File: src/HE.InvestmentLoans.BusinessLogic/User/LoanUserContext.cs (hunk 1 above)
Comment (lukasz-ciura-pwc) [inline src/HE.InvestmentLoans.BusinessLogic/User/LoanUserContext.cs:30]: Please add Unit Tests.

**Review by lukasz-ciura-pwc — CHANGES_REQUESTED** [review]

File: src/HE.InvestmentLoans.BusinessLogic/User/LoanUserContext.cs (hunk 1 above)
Comment (dawid-pawulski) [reply src/HE.InvestmentLoans.BusinessLogic/User/LoanUserContext.cs:30]: There is seperate task for unit tests in this story.
They will be added in another pr

**Comment by dawid-pawulski** [comment]
Created new one PR here:
https://github.com/homesengland/he-ssp/pull/223

**Pull request closed without merging**